core/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -209,14 +209,20 @@ public void onFailure(Exception replicaException) {
                     shardReplicaFailures.add(new ReplicationResponse.ShardInfo.Failure(
                         shard.shardId(), shard.currentNodeId(), replicaException, restStatus, false));
                     String message = String.format(Locale.ROOT, "failed to perform %s on replica %s", opType, shard);
+                    // TODO: Boaz wanted to get rid of this warning, should we?
                     logger.warn(
```

Comment:
we plan to use this for background operations like the global check point sync, where it's not a big deal if this happen. I don't want the logs to have warning in them. Instead, implementations (i.e., TransportWriteAction) of `failShardIfNeeded` can log a warning if they are going to fail the shard